src/org/labkey/test/components/ui/navigation/NavBar.java

```diff
@@ -117,5 +125,13 @@ protected abstract class ElementCache extends Component<ElementCache>.ElementCac
         public WebElement projectNameDisplay = Locator.tagWithClass("span", "project-name").findWhenNeeded(this);
         public Input searchBox = Input.Input(Locator.tagWithClass("input", "navbar__search-input"), getDriver()).findWhenNeeded(this);
         public MultiMenu searchMenu = new MultiMenu.MultiMenuFinder(getDriver()).withButtonId("find-and-search-menu").findWhenNeeded(this);
+        public final MultiMenu folderMenu = new MultiMenu.MultiMenuFinder(getDriver())
+                .withButtonId("folder-menu").refindWhenNeeded(getDriver());
+        public final ProductMenu productMenu(int columnCount)
+        {
+            return ProductMenu.finder(getDriver()).timeout(1000).findWhenNeeded(this)
+                    .setExpectedSectionCount(columnCount);
+        }
```

Comment:
I'd avoid exposing a public method that forces one to specify the expected column count (`getProductMenu`, earlier in this PR). You could get the count from the SM/BioNavBar subclasses (instead of getting the menu itself like this used to do)
```suggestion
        public final ProductMenu productMenu = ProductMenu.finder(getDriver()).timeout(1000).findWhenNeeded(this)
                .setExpectedSectionCount(getExpectedProductMenuColumnCount());
```

Then add an abstract `getExpectedProductMenuColumnCount` to the `NavBar` or the `NavBar.ElementCache`:
```
public abstract int getExpectedProductMenuColumnCount();
```

Reply:
The count is no longer needed.